### PR TITLE
Siax Combat Replay

### PR DESCRIPTION
**Pull request by Linkaaaaa**
Added another ID for the Caustic Explosion phase to the mechanics.
Added various other Skill IDs.
Bonus: Added a wind-up animation for MAMA's knights knockback AoE.

Vile Spit:

Nightmare Hallucinations Spawn AoE:

Caustic Explosion (Breakbar)

Caustic Explosion (66% & 33%)

Caustic Explosion (Echos)

Tail Lash

Caustic Barrage

Cascade Of Torment

Volatile Expulsion

### CODE REVIEW

File: GW2EIEvtcParser/EncounterLogic/Fractals/Nightmare/Siax.cs [hunk 1]
```diff
@@ -99,5 +100,148 @@ internal override List<PhaseData> GetPhases(ParsedEvtcLog log, bool requirePhase
...
+                        int attackEnd = (int)c.Time + duration;
+                        int stunTime = attackEnd;
+                        int detTime = attackEnd;
+
+                        IReadOnlyDictionary<long, BuffsGraphModel> bgmUptime = target.GetBuffGraphs(log);
+
+                        BuffsGraphModel bgmStun = bgmUptime.TryGetValue(Stun, out BuffsGraphModel value) ? value : null;
+                        if (bgmStun != null)
+                        {
+                            Segment segment = bgmStun.BuffChart.FirstOrDefault(x => x.Start > c.Time && x.Start < c.Time + duration);
```
Comment (EliphasNUIT) [inline GW2EIEvtcParser/EncounterLogic/Fractals/Nightmare/Siax.cs:126]: You should be checking for x.Value, the segment returned could very well a period of time where the target was not stunned. Same for the others below. You can also use the "GetBuffStatus" API on target to reduce quite a bit of code regarding stun checking.

File: GW2EIEvtcParser/EncounterLogic/Fractals/Nightmare/Siax.cs [hunk 2]
```diff
@@ -99,5 +100,198 @@ internal override List<PhaseData> GetPhases(ParsedEvtcLog log, bool requirePhase
...
+                            attackEnd = Math.Min((int)detSegment.End, attackEnd); // End of determinated
+                        }
+
+                        replay.Decorations.Add(new CircleDecoration(true, expectedHitTime, 1500, (start, attackEnd), "rgba(255, 0, 0, 0.2)", new AgentConnector(target)));
+                        replay.Decorations.Add(new CircleDecoration(true, 0, 1500, (start, attackEnd), "rgba(255, 0, 0, 0.2)", new AgentConnector(target)));
+                    }
+                    // Poison AoE
+                    var vileSpit = casts.Where(x => x.SkillId == VileSpit).ToList();
+                    EffectGUIDEvent poisonField = log.CombatData.GetEffectGUIDEvent(EffectGUIDs.VileSpitSiax);
+                    if (vileSpit.Count > 0 && poisonField != null)
```
Comment (EliphasNUIT) [inline GW2EIEvtcParser/EncounterLogic/Fractals/Nightmare/Siax.cs:163]: Just for information, what is the reasoning behind the cout check?

Comment (Linkaaaaa) [reply GW2EIEvtcParser/EncounterLogic/Fractals/Nightmare/Siax.cs:163]: nothing specific, was a double measure to check that the skill got casted other than just the event, can be removed if you want

Comment (EliphasNUIT) [reply GW2EIEvtcParser/EncounterLogic/Fractals/Nightmare/Siax.cs:163]: Yeah, if they are not useful, they should be removed.

**Comment by EliphasNUIT** [comment]
Thanks!

**Pull request merged**